src/analyses/basePriv.ml

```diff
@@ -0,0 +1,1664 @@
+
+  val escape: Q.ask -> (varinfo -> G.t) -> (varinfo -> G.t -> unit) -> BaseComponents (D).t -> EscapeDomain.EscapedVars.t -> BaseComponents (D).t
+  val enter_multithreaded: Q.ask -> (varinfo -> G.t) -> (varinfo -> G.t -> unit) -> BaseComponents (D).t -> BaseComponents (D).t
+  val threadenter: Q.ask -> BaseComponents (D).t -> BaseComponents (D).t
+
+  (* TODO: better name *)
+  val is_private: Q.ask -> varinfo -> bool
+
+  val init: unit -> unit
+  val finalize: unit -> unit
+end
+
+module NoInitFinalize =
+struct
+  let init () = ()
+  let finalize () = ()
+end
+
+let old_threadenter (type d) ask (st: d BaseDomain.basecomponents_t) =
+  let new_cpa = if not (!GU.earlyglobs || ThreadFlag.is_multi ask) then CPA.filter_class 2 st.cpa else CPA.filter (fun k v -> Basetype.Variables.is_global k) st.cpa in
+  {st with cpa = new_cpa}
+
+let startstate_threadenter (type d) (startstate: unit -> d) ask (st: d BaseDomain.basecomponents_t) =
+  {st with cpa = CPA.bot (); priv = startstate ()}
+
+module OldPrivBase =
+struct
+  include NoInitFinalize
+  module D = Lattice.Unit
+
+  let startstate () = ()
+
+  let lock ask getg cpa m = cpa
+  let unlock ask getg sideg st m = st
+
+  let escape ask getg sideg st escaped = st
+  let enter_multithreaded ask getg sideg st = st
+  let threadenter = old_threadenter
+
+  let sync_privates reason ask =
+    match reason with
+    | `Init
+    | `Thread ->
+      true
+    | _ ->
+      !GU.earlyglobs && not (ThreadFlag.is_multi ask)
+end
+
+(* Copy of OldPriv with is_private constantly false. *)
+module NoPriv: S =
+struct
+  include OldPrivBase
+
+  module G = BaseDomain.VD
+
+  let read_global ask getg (st: BaseComponents (D).t) x =
+    getg x
+
+  let write_global ask getg sideg (st: BaseComponents (D).t) x v =
+    (* Here, an effect should be generated, but we add it to the local
+     * state, waiting for the sync function to publish it. *)
+    (* Copied from MainFunctor.update_variable *)
+    if ((get_bool "exp.volatiles_are_top") && (is_always_unknown x)) then
+      {st with cpa = CPA.add x (VD.top ()) st.cpa}
+    else
+      {st with cpa = CPA.add x v st.cpa}
+
+  let is_private (a: Q.ask) (v: varinfo): bool = false
+
+  let sync ask getg (st: BaseComponents (D).t) reason =
+    (* For each global variable, we create the diff *)
+    let add_var (v: varinfo) (value) ((st: BaseComponents (D).t),acc) =
+      if M.tracing then M.traceli "globalize" ~var:v.vname "Tracing for %s\n" v.vname;
+      let res =
+        if is_global ask v then begin
+          if M.tracing then M.tracec "globalize" "Publishing its value: %a\n" VD.pretty value;
+          ({st with cpa = CPA.remove v st.cpa}, (v,value) :: acc)
+        end else
+          (st,acc)
+      in
+      if M.tracing then M.traceu "globalize" "Done!\n";
+      res
+    in
+    (* We fold over the local state, and collect the globals *)
+    CPA.fold add_var st.cpa (st, [])
+end
+
+module OldPriv: S =
```

Comment:
Maybe one should remark somewhere that it is unsound?!
And maybe even remove it from the official list of options for this listed in `defaults.ml`

Reply:
I'm doing the renaming of all of these locally and have added a comment here about its unsoundness.